### PR TITLE
Implement auth button A / B test

### PR DESCRIPTION
**Pull request by srt32**
Configures https://github.com/andrew/split and sets up an A/B test on the text
on the authenticate button.

screen shot 2015-03-03 at 11 12 28 am
screen shot 2015-03-03 at 11 12 33 am

https://trello.com/c/qmurdq4Z/530-implement-a-b-testing-for-homepage

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ en:
   sync_repos: "Refresh repo list"
   syncing_repos: "Loading repos..."
   authenticate: "Sign In with GitHub"
+  aggressive_authenticate: "Get Hounding!"
```
Comment (srt32) [inline config/locales/en.yml]: Thoughts on some copy we actually want to test?

File: app/views/application/_segment_io.haml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 :javascript
   window.analytics||(window.analytics=[]),window.analytics.methods=["identify","track","trackLink","trackForm","trackClick","trackSubmit","page","pageview","ab","alias","ready","group","on","once","off"],window.analytics.factory=function(t){return function(){var a=Array.prototype.slice.call(arguments);return a.unshift(t),window.analytics.push(a),window.analytics}};for(var i=0;i<window.analytics.methods.length;i++){var method=window.analytics.methods[i];window.analytics[method]=window.analytics.factory(method)}window.analytics.load=function(t){var a=document.createElement("script");a.type="text/javascript",a.async=!0,a.src=("https:"===document.location.protocol?"https://":"http://")+"d2dq2ahtl5zl1z.cloudfront.net/analytics.js/v1/"+t+"/analytics.min.js";var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(a,n)},window.analytics.SNIPPET_VERSION="2.0.8",
   window.analytics.load("#{ENV['SEGMENT_IO_WRITE_KEY']}");
-  window.analytics.page("#{yield :page_title}");
+  window.analytics.page("#{[yield(:page_title), yield(:variation)].compact.uniq.join(" > ")}");
```
Comment (srt32) [inline app/views/application/_segment_io.haml]: @graysonwright I cargo culted a fair amount of your work from a closed PR. Thanks!

File: config/locales/en.yml (hunk 1 above)
Comment (teoljungberg) [inline config/locales/en.yml]: `Get started with Hound` ?

File: app/controllers/home_controller.rb [hunk 3]
```diff
@@ -4,6 +4,7 @@ class HomeController < ApplicationController
   before_action :redirect_to_repos, if: :signed_in?
 
   def index
+    @button_type = ab_test("auth_button")
```
Comment (salbertson) [inline app/controllers/home_controller.rb]: Does `@auth_button_text` make more sense?

File: config/locales/en.yml (hunk 1 above)
Comment (salbertson) [inline config/locales/en.yml]: Start using Hound today!

Comment (salbertson) [inline config/locales/en.yml]: Get started

Comment (salbertson) [inline config/locales/en.yml]: Try out Hound today!

File: app/controllers/home_controller.rb (hunk 3 above)
Comment (srt32) [inline app/controllers/home_controller.rb]: It's not the actual text here. It's either `"original"` or `"aggressive"`. The value of the alternatives gets evaluated in the view.

File: config/locales/en.yml (hunk 1 above)
Comment (srt32) [inline config/locales/en.yml]: I like `Try out Hound today!`. Updated in 07afa19. Open to other suggestions as well.

Comment (salbertson) [inline config/locales/en.yml]: Have your code reviewed by a trusty Hound

File: app/controllers/home_controller.rb (hunk 3 above)
Comment (salbertson) [inline app/controllers/home_controller.rb]: Ah, ok.

File: config/locales/en.yml [hunk 4]
```diff
@@ -2,6 +2,7 @@ en:
   sync_repos: "Refresh repo list"
   syncing_repos: "Loading repos..."
   authenticate: "Sign In with GitHub"
+  aggressive_authenticate: "Try out Hound today!"
   active_repos: "Active repos"
   search_placeholder: "Search by repo"
   pending_status: "Hound is busy reviewing changes..."
```
Comment (croaky) [inline config/locales/en.yml]: `:sort`?

File: app/views/application/_segment_io.haml (hunk 2 above)
Comment (croaky) [inline app/views/application/_segment_io.haml]: I think we want the same event ("Viewed Home Page") but with alternate properties (`cta_text: "Sign in with GitHub"` or `cta_text: "Get Hounding"`)

**Comment by croaky** [comment]
Where is the text set? Redis? Can you document how to create variations?

**Comment by salbertson** [comment]
@srt32, when you have a chance would you give an update for this?

**Comment by srt32** [comment]
@croaky the configs are stored here: https://github.com/thoughtbot/hound/blob/st-configure-ab-test/config/initializers/split.rb

@salbertson some remaining questions are whether we want to use Segment to report out the test results (the current implementation) or use the [web interface](https://github.com/andrew/split#web-interface) Split already provides. Thoughts?

**Comment by salbertson** [comment]
@srt32, I think it makes sense to use Segment. @croaky?

**Comment by salbertson** [comment]
@srt32 @croaky any update on this?

**Comment by croaky** [comment]
Yes, Segment and Split dashboard. See FormKeep for inspiration:

https://github.com/thoughtbot/formkeep/commit/e30c73d7da7082134062d0fceb75c28e0a5bf31f

https://github.com/thoughtbot/formkeep/commit/f660f2dcb2b218e84ccbdc77bd7af91ccabc2a70

**Comment by srt32** [comment]
Thanks! Closing in favor of https://github.com/thoughtbot/hound/pull/746

**Pull request closed without merging**